controllers/main.js

```diff
@@ -123,28 +121,43 @@ module.exports.renderPlay = async (req, res) => {
 module.exports.renderResults = async (req, res, next) => {
   const { id } = req.params;
   let playSessionData = await PlaySession.findOne({ sessionId: id });
-  const highscoreText = `${
-    remapDifficultyTexts[playSessionData.difficulty] + playSessionData.length
-  }`;
-  const difficultyFilter = `bestScores.${playSessionData.difficulty}`;
-  const highscoresData = JSON.stringify(
-    await User.find()
-      .sort({ [difficultyFilter]: -1 })
-      .limit(10)
-  );
 
   playSessionData = JSON.stringify(playSessionData);
   res.render("main/results", {
     extraStyles: resultsStyle + mediaButtonsStyle,
     playSessionData,
-    highscoresData,
-    highscoreText,
-    defaultProfileImg: playSettingsImage,
     sessionId: id,
   });
   next();
 };
 
+module.exports.fetchTopHighscores = async (req, res) => {
+  const { difficulty, length } = req.query;
+  const dbSearchFilter = `bestScores.${difficulty}.${length}`;
+  async function getTop10Users() {
```

Comment:
Can we have this function moved outside `fetchTopHighscores`?

Reply:
Think this bit of code technically shouldn't be relevant to this PR request, but i think i rebased the branch poorly, and it shows the changes from a previous, already merged PR. 

I did make the suggested change though.